### ISSUE TITLE
[openSUSE] basic config settings for Postfix on Rockstor install stage

### ISSUE THREAD

**Issue by tyukh** [rockstor/rockstor-core#2132]
Config **/etc/postfix/main.cf** need modification from default value `inet_protocols = all` to `inet_protocols = ipv4` in case disabling IPV6 in NetworkManager and error in saving alert mail setting in Rockstor GUI. 
Config **/etc/postfix/master.cf** need modification from commented default value `#tlsmgr unix - - n 1000? 1 tlsmg`  to `tlsmgr unix - - n 1000? 1 tlsmg` in case using SMTP TLS encryption for such services as gmail.com or outlook.com

**Comment by phillxnet** [comment]
@tyukh Thanks for this. Nice to have an issue submitted with included fix.

If you fancy having a go at either of your issues or any others then do make a quick note on the relevant issue to avoid duplicated effort. I say this as given the relevance of these to our openSUSE move I may well have a go at them myself. I will likewise make a note if I get started on either of them.

**Comment by phillxnet** [comment]
@tyukh As this issue relates to our programmatic editing of /etc/postfix/main.cf, which is in common with your other issue that I am currently working on I'm going to take a look at this issue concurrently.

The issues are definitely properly separate but there treatment with rockstor code is close enough for me to sensibly, I hope, work on them in parallel. This may save me some merge conflicts down the line, and help with reduced overhead on testing them independently and preparing their respective pull requests.

Linking to related issue on our need to edit /etc/postfix/master.cf contextually re our distro base:

[openSUSE] fix cert file path in alert mail config #2133

**Comment by phillxnet** [comment]
Issue notes:
Changes to master.cf require a subsequent "postfix reload".

**Comment by tyukh** [comment]
> @tyukh As this issue relates to our programmatic editing of /etc/postfix/main.cf, which is in common with your other issue that I am currently working on I'm going to take a look at this issue concurrently.
I split this issue with #2133 cause i see three different solutions depends of project strategy in packaging and delivering. It can be solved 1. in basic iso image, or 2. in Rockstor install script, or 3. in runtime email alert setup script. Probably the third variant is the simplest at the moment, but probably 1 and 2 also deserve attention for strategic reasons.

**Comment by phillxnet** [comment]
@tyukh Thanks for re-opening this issue as I would like to use it as the main one given it pertains to the configuration and we also have the issue of sysconfig potentially overriding us here given we edit these files in code.

I've already begun working on this issue and have made some headway under a branch similar to this issue, hence the re-opening.

**Comment by tyukh** [comment]
> @tyukh Thanks for re-opening this issue as I would like to use it as the main one given it pertains to the configuration and we also have the issue of sysconfig potentially overriding us here given we edit these files in code.
> 
> I've already begun working on this issue and have made some headway under a branch similar to this issue, hence the re-opening.
Sorry, it was github' newbie mistake

**Comment by phillxnet** [comment]
I think for the time being, re the various approaches, I favour 3 also currently and that is the way I'm going.

The proposed installer is JeOS based but we are really trying to maintain compatibility with a lightly config modified Leap15.1 hence we can't approach this only in the ISO as not vanilla Leap installs will then work.

I'm not favouring the 'instal script' as we have to modify / re-assert email setting on every re-config, so we are king of stuck with live edits after each config is my thinking.

And I think for simplicity and to get to our feature parity as quickly as possible we will just have to deactivate sysconfig's email configuration, ie I believe the default is:
```
MAIL_CREATE_CONFIG="yes"
```
Which means, by default, any changes we make directly to main.cf will be overwritten by the sysconfig system on each and every reboot. Have you found this in your setup. I'm still at the early stages of looking at our options here.

**Comment by tyukh** [comment]
By the way, I came across a problem with mail alerts when I received an error in viewing the SMART (bad) state of one of my disks: since I didn't have mail alerts configured I received a script error GUI pane. So another issue - checking is email alert is configured before trying to send email warning of SMART status.

**Comment by phillxnet** [comment]
> So another issue - checking is email alert is configured before trying to send email warning of SMART status.
...
> I received a script error GUI pane.

I'm not so sure of that as this didn't happen in our CentOS base. And we can't require folks to enable and configure email notification, so all stuff should just be emailed to root in that case. So yes, it would be good to have that as another issue with all relevant details. But I can't imagine that openSUSE default config doesn't email root issues as they arrise. Maybe it was down to how our config is interacting differently with their base config. Email is tricky and we won't fix these all in one issue so yes, create another issue with your findings and how you reproduced that GUI panel error. Also best if you can do it from a default Leap15.1 config (with our minimal changes as per the wiki etc). Cheers.

**Comment by tyukh** [comment]
> I'm not so sure of that as this didn't happen in our CentOS base. And we can't require folks to enable and configure email notification, so all stuff should just be emailed to root in that case. So yes, it would be good to have that as another issue with all relevant details. But I can't imagine that openSUSE default config doesn't email root issues as they arrise. Maybe it was down to how our config is interacting differently with their base config. Email is tricky and we won't fix these all in one issue so yes, create another issue with your findings and how you reproduced that GUI panel error. Also best if you can do it from a default Leap15.1 config (with our minimal changes as per the wiki etc). Cheers.

I have reproduce error in SMART. Postfix failed to start cause `inet_protocols = all` so it's mostly OpenSUSE specific and linked to that issue. Here log:

```
Traceback (most recent call last):
  File "/opt/rockstor/src/rockstor/rest_framework_custom/generic_view.py", line 41, in _handle_exception
    yield
  File "/opt/rockstor/src/rockstor/storageadmin/views/disk_smart.py", line 116, in post
    return self._info(disk)
  File "/opt/rockstor/eggs/Django-1.8.16-py2.7.egg/django/utils/decorators.py", line 145, in inner
    return func(*args, **kwargs)
  File "/opt/rockstor/src/rockstor/storageadmin/views/disk_smart.py", line 62, in _info
    e_summary, e_lines = error_logs(disk.name, disk.smart_options)
  File "/opt/rockstor/src/rockstor/system/smart.py", line 199, in error_logs
    screen_return_codes(e_msg, overide_rc, o, e, rc, smart_command)
  File "/opt/rockstor/src/rockstor/system/smart.py", line 409, in screen_return_codes
    email_root("S.M.A.R.T error", msg_on_hit)
  File "/opt/rockstor/src/rockstor/system/email_util.py", line 76, in email_root
    smtp = smtplib.SMTP('localhost')
  File "/usr/lib64/python2.7/smtplib.py", line 256, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/lib64/python2.7/smtplib.py", line 317, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/lib64/python2.7/smtplib.py", line 292, in _get_socket
    return socket.create_connection((host, port), timeout)
  File "/usr/lib64/python2.7/socket.py", line 575, in create_connection
    raise err
error: [Errno 99] Cannot assign requested address
```

**Comment by phillxnet** [comment]
@tyukh Re:

> I have reproduce error in SMART view. Postfix failed to start cause `inet_protocols = all` so it's mostly OpenSUSE specific and linked to that issue.

That's good to know, Thanks. So another knock-on of our ipv4 only Network Manager setting then.
Cheers.

**Comment by phillxnet** [comment]
I have confirmed that a default sysconfig re-writes our mail config on a JeOS install; post edit we have:
```
cat /etc/postfix/main.cf | grep inet_protocol
inet_protocols = ipv4
config.postfix
Reading /etc/sysconfig and updating the system...
Setting up postfix local as MDA...
Setting SPAM protection to "off"...
```
and we have a reversal of our config to the openSUSE sysconfig defined defaults.
```
cat /etc/postfix/main.cf | grep inet_protocol
inet_protocols = all
```
Editing /etc/sysconfig/mail and changing:
```
MAIL_CREATE_CONFIG="yes"
```
to
```
MAIL_CREATE_CONFIG="no"
```
and our config.postfix leaves our settings alone there after.
```
i.e.:
```
cat /etc/postfix/main.cf | grep inet_protocol
inet_protocols = ipv4
config.postfix
Reading /etc/sysconfig and updating the system...
cat /etc/postfix/main.cf | grep inet_protocol
inet_protocols = ipv4
```

**Comment by phillxnet** [comment]
Linking this issue to to a pre-existing issue re the /etc/sysconfig/mail mechanism discussed earlier:

[openSUSE] disable config.postfix as we edit main.cf directly #2035

I have in turn edited the pending pr as closing #2035 as well.
